target/linux/bcm53xx/patches-5.15/180-usb-xhci-add-support-for-performing-fake-doorbell.patch

```diff
@@ -90,41 +90,22 @@ it on BCM4708 family.
  /*
   * Reset a halted HC.
   *
-@@ -611,10 +654,20 @@ static int xhci_init(struct usb_hcd *hcd
- 
- static int xhci_run_finished(struct xhci_hcd *xhci)
- {
--	if (xhci_start(xhci)) {
--		xhci_halt(xhci);
--		return -ENODEV;
-+	int err;
-+
-+	err = xhci_start(xhci);
-+	if (err) {
-+		err = -ENODEV;
-+		goto err_halt;
+@@ -634,6 +677,15 @@ static int xhci_run_finished(struct xhci
+ 		spin_unlock_irqrestore(&xhci->lock, flags);
+ 		return -ENODEV;
  	}
 +
 +	if (xhci->quirks & XHCI_FAKE_DOORBELL) {
-+		err = xhci_fake_doorbell(xhci, 1);
-+		if (err)
-+			goto err_halt;
++		int err = xhci_fake_doorbell(xhci, 1);
++		if (err) {
++			xhci_halt(xhci);
```

Comment:
It seems that we need `spin_unlock_irqrestore(&xhci->lock, flags);` here? Not sure. Same on kernel 6.1.

Reply:
I cannot say... someone with more knowledge will need to help out.

Reply:
Maybe ping Rafał Miłecki?

Reply:
@rmilecki - can you help out with this rebase?

Reply:
You probably wanted to ping @rmilecki 

The patch uses usleep_range(), you can not call that in atomic context like a spinlock.

Reply:
Rafal mentioned in personal email that he was quite busy but will try to get to this sometime this week.  Meanwhile, I pushed the 5.15.137 update.